cmd/policy-assistant/pkg/matcher/traffic.go

```diff
@@ -71,10 +75,245 @@ func (p *TrafficPeer) IsExternal() bool {
 	return p.Internal == nil
 }
 
+func (p *TrafficPeer) Translate() TrafficPeer {
+	var podsNetworking []*PodNetworking
+	var podLabels map[string]string
+	var namespaceLabels map[string]string
+	var workloadOwner string
+	var internalPeer InternalPeer
+	workloadOwnerExists := false
+	workloadMetadata := strings.Split(strings.ToLower(p.Internal.Workload), "/")
+	if len(workloadMetadata) != 3 || (workloadMetadata[0] == "" || workloadMetadata[1] == "" || workloadMetadata[2] == "") || (workloadMetadata[1] != "daemonset" && workloadMetadata[1] != "statefulset" && workloadMetadata[1] != "replicaset" && workloadMetadata[1] != "deployment" && workloadMetadata[1] != "pod") {
+		logrus.Fatalf("Bad Workload structure: Types supported are pod, replicaset, deployment, daemonset, statefulset, and 3 fields are required with this structure, <namespace>/<workloadType>/<workloadName>")
+	}
+	kubeClient, err := kube.NewKubernetesForContext("")
+	utils.DoOrDie(err)
+	ns, err := kubeClient.GetNamespace(workloadMetadata[0])
+	utils.DoOrDie(err)
+	kubePods, err := kube.GetPodsInNamespaces(kubeClient, []string{workloadMetadata[0]})
+	if err != nil {
+		logrus.Fatalf("unable to read pods from kube, ns '%s': %+v", workloadMetadata[0], err)
+	}
+	for _, pod := range kubePods {
+		if workloadMetadata[1] == "deployment" && pod.OwnerReferences[0].Kind == "ReplicaSet" {
+			kubeReplicaSets, err := kubeClient.GetReplicaSet(workloadMetadata[0], pod.OwnerReferences[0].Name)
+			if err != nil {
+				logrus.Fatalf("unable to read Replicaset from kube, rs '%s': %+v", pod.OwnerReferences[0].Name, err)
+			}
+			workloadOwner = kubeReplicaSets.OwnerReferences[0].Name
+		} else if workloadMetadata[1] == "daemonset" || workloadMetadata[1] == "statefulset" || workloadMetadata[1] == "replicaset" {
+			workloadOwner = pod.OwnerReferences[0].Name
+		} else if workloadMetadata[1] == "pod" {
+			workloadOwner = pod.Name
+		}
+		if workloadOwner == workloadMetadata[2] {
+			podLabels = pod.Labels
+			namespaceLabels = ns.Labels
+			podNetworking := PodNetworking{
+				IP: pod.Status.PodIP,
+			}
+			podsNetworking = append(podsNetworking, &podNetworking)
+			workloadOwnerExists = true
+
+		}
+	}
+
+	if !workloadOwnerExists {
+		logrus.Infof("workload not found on the cluster")
+		internalPeer = InternalPeer{
+			Workload: "",
```

Comment:
hey @huntergregory , thanks for the review, the problem here is that doing a fatal exits my code when a workload is scaled down to 0. for example if you have a deployment with 0 replicas it crashes there when you call the function to map all the deployments on the cluster to trafficpeers. Have you thought about that scenario.

Reply:
My suggestion is nitpicky, and I'm noticing it requires a good amount of change, so honestly feel free to leave everything as is 🙂. I was more so suggesting that we could change the function signature to:
```go
func (p *TrafficPeer) Translate() (*TrafficPeer, error) {
```

and instead of using `logrus.Fatal()`, we could start returning errors to be handled as needed:
```go
return nil, fmt.Errorf("failed to get workload: %w", err)
```

But this might be a lot of work for little reward: if you replace `logrus.Fatal()` in this function, you might as well replace it in all functions. So feel free to leave as is 🙂

Reply:
> for example if you have a deployment with 0 replicas

Good call out. Do you mind calling this out somehow in the code too?